test/cloudtest/test_storage.py

```diff
@@ -94,11 +128,15 @@ def test_source_resizing(mz: MaterializeApplication) -> None:
         no_reset=True,
     )
 
-    wait(condition="condition=Ready", resource=storage_cluster)
+    wait(
+        condition="condition=Ready",
+        resource="pod",
+        label="cluster.environmentd.materialize.cloud/workers=16",
+    )
 
-    # NB: We'd like to do the following, but jsonpath gives us no way to express it!
-    # TODO: revisit or handroll a retry loop
-    # wait(condition=f"jsonpath=metadata.labels.storage.environmentd.materialize.cloud/size=16", resource=storage_cluster)
+    assert get_cluster_workers(mz, cluster_id) == "'16'"
```

Comment:
This test is apparently flickering. The result can be either `'16'` or `'1 16'` depending on whether the previous pod has already been destroyed.

Any opinions on how to address this, @philip-stoev, @def-? Is there an easy way to await the old pod being destroyed or shall I do the comparison using `endswith`?

Reply:
You can wait for a given old pod to go away:

```
wait(condition="delete", resource="pod/...")
```

The only thing is to figure out the name of the pod, you need to do that before the `ALTER ` and keep it and use it in the `wait` after the `ALTER`. 

`wait()` calls `kubectl wait` which has this `delete` condition as a special-case.